File: intellex-frontend/frontend/src/lib/api/analytics.ts
import { apiClient } from "@/lib/api/client";
import type { PipelineStats } from "@/lib/types";

interface RawPipelineStats {
  total_documents: number;
  total_events: number;
  total_sources: number;
  sources: string[];
  last_run_at: string | null;
  last_run_fetched: number;
  last_run_unique: number;
  dedup_rate: number;
  refresh_interval_minutes: number;
  is_running: boolean;
}

export async function getPipelineStats(): Promise<PipelineStats> {
  const { data } = await apiClient.get<RawPipelineStats>(
    "/analytics/pipeline"
  );

  return {
    totalDocuments: data.total_documents,
    totalEvents: data.total_events,
    totalSources: data.total_sources,
    sources: data.sources,
    lastRunAt: data.last_run_at,
    lastRunFetched: data.last_run_fetched,
    lastRunUnique: data.last_run_unique,
    dedupRate: data.dedup_rate,
    refreshIntervalMinutes: data.refresh_interval_minutes,
    isRunning: data.is_running,
  };
}

File: intellex-backend/backend/app/api/schemas/analytics.py
from datetime import datetime

from pydantic import BaseModel


class PipelineStatsResponse(BaseModel):
    total_documents: int

    total_events: int

    total_sources: int

    sources: list[str]

    last_run_at: datetime | None

    last_run_fetched: int

    last_run_unique: int

    dedup_rate: float

    refresh_interval_minutes: int

    is_running: bool
